### PR TITLE
docs(SDB): Update connection page and troubleshooting MTA-4792

### PR DESCRIPTION
**Pull request by SamyOubouaziz**
### Your checklist for this pull request

- [ ] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description

Please describe what you added or changed.

### CODE REVIEW

File: serverless/sql-databases/how-to/connect-to-a-database.mdx [hunk 1]
```diff
@@ -15,6 +15,28 @@ categories:
 
 This page shows you how to set up the connection to a Serverless SQL Database using the Scaleway console.
 
+## Overview
+
+To connect to a Serverless SQL Database, you can either use a **connection string**, or **connection parameters**.
+
+  - A connection string provides the necessary information and parameters to establish a connection between an IAM principal (i.e. a user or an application) and the database. The string is written as follows:
```
Comment (fpagny) [inline serverless/sql-databases/how-to/connect-to-a-database.mdx:22]:
```suggestion
  - A connection string provides the necessary information and parameters to establish a connection between an [IAM User](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#user) or [Application](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#application) and the database. The string is written as follows:
```
IAM Principal includes also IAM groups, which cannot cannot to a database. Hence I recall we settled on the standard formulation "IAM User or Application"

File: serverless/sql-databases/how-to/connect-to-a-database.mdx [hunk 2]
```diff
@@ -15,6 +15,28 @@ categories:
...
+      ```
+      postgres://[user-or-application-id]:[api-secret-key]@[database-hostname]:5432/[database-name]?sslmode=require
+      ```
+      <Message type="note">
+      The `?sslmode=require` is mandatory to connect to your database. Refer to the [dedicated documentation](/serverless/sql-databases/api-cli/secure-connection-ssl-tls/) for more information about SSL connection to Serverless SQL Databases.
+      </Message>
+
+  - Connection parameters provide the necessary information and parameters to connect an IAM principal (i.e. a user or an application) to a database. The parameters are expressed in the `KEY="value"` value format, as follows:
```
Comment (fpagny) [inline serverless/sql-databases/how-to/connect-to-a-database.mdx:30]:
```suggestion
  - Connection parameters provide the necessary information and parameters to connect an[IAM User](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#user) or [Application](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#application) to a database. The parameters are expressed in the `KEY="value"` value format, as follows:
```

File: serverless/sql-databases/how-to/connect-to-a-database.mdx (hunk 1 above)
Comment (nerda-codes) [inline serverless/sql-databases/how-to/connect-to-a-database.mdx:22]: Maybe link to the principal concept

File: serverless/sql-databases/how-to/connect-to-a-database.mdx [hunk 3]
```diff
@@ -15,6 +15,28 @@ categories:
...
+
+      ```
+      PGUSER="user-or-application-id"
+      PGPASSWORD="api-secret-key"
+      PGHOST="database-hostname"
+      PGPORT="5432"
+      PGDATABASE="database-name"
+      ```
+
```
Comment (fpagny) [inline serverless/sql-databases/how-to/connect-to-a-database.mdx:39]:
```suggestion
      <Message type="note">
      The `PGSSLMODE="require"` is mandatory to connect to your database but should be the default value with most SQL clients. Refer to the [dedicated documentation](/serverless/sql-databases/api-cli/secure-connection-ssl-tls/) for more information about SSL connection to Serverless SQL Databases.
      </Message>
```

File: serverless/sql-databases/how-to/connect-to-a-database.mdx [hunk 4]
```diff
@@ -77,4 +121,6 @@ Use the connection parameters (host, port, user, password, and database name) pr
   Connection methods may vary depending on the framework or library used in your application. Refer to the corresponding official documentation for more information.
```
Comment (fpagny) [inline serverless/sql-databases/how-to/connect-to-a-database.mdx:121]: We can also point here to the documentation about SSL/TLS: https://www.scaleway.com/en/docs/serverless/sql-databases/api-cli/secure-connection-ssl-tls/
It's about SSL/TLS but also provide code snippets for many clients and languages that could help.

**Review by fpagny — DISMISSED** [review]

File: serverless/sql-databases/how-to/connect-to-a-database.mdx (hunk 1 above)
Comment (RoRoJ) [reply serverless/sql-databases/how-to/connect-to-a-database.mdx:22]: :+1: but user and application should not be capitalized (to stay consistent with existing doc)

File: serverless/sql-databases/troubleshooting/connection-errors.mdx [hunk 5]
```diff
@@ -13,6 +13,45 @@ categories:
...
+
+The error message below appears when trying to connect to the Serverless SQL Database:
+
+    ```
+    FATAL: password authentication failed for user "example-fafb-4c6c-a406-911d9675a703"
+    ```
+
+### Cause
+
+You either used a wrong password/IAM secret key, or a wrong username/IAM user ID/application ID.
```
Comment (RoRoJ) [inline serverless/sql-databases/troubleshooting/connection-errors.mdx:28]:
```suggestion
You either used the wrong password/IAM secret key, or the wrong username/IAM user ID/application ID.
```

File: serverless/sql-databases/troubleshooting/connection-errors.mdx [hunk 6]
```diff
@@ -13,6 +13,45 @@ categories:
...
+
+The error message below appears when trying to connect to the Serverless SQL Database:
+
+    ```
+    FATAL: principal example-fafb-4c6c-a406-911d9675a703 does not have permission to access database example-a2ea-4f6e-bbe5-791a1b6905ff
+    ```
+
+### Cause
+
+The [IAM principal](/identity-and-access-management/iam/concepts/#principal) (User or application) does not have the necessary permissions to connect the database.
```
Comment (RoRoJ) [inline serverless/sql-databases/troubleshooting/connection-errors.mdx:46]:
```suggestion
The [IAM principal](/identity-and-access-management/iam/concepts/#principal) (user or application) does not have the necessary permissions to connect the database.
```

**Review by jcirinosclwy — APPROVED** [review]

**Review by nerda-codes — APPROVED** [review]

**Pull request merged**